### ISSUE TITLE
Crash when using top level init

### ISSUE THREAD

**Issue by Nokel81** [getsentry/sentry-electron#456]
- [x] Review the documentation: https://docs.sentry.io/platforms/javascript/electron/
- [x] Search for existing issues: https://github.com/getsentry/sentry-electron/issues
- [x] Use the latest release: https://github.com/getsentry/sentry-electron/releases
- [ ] ~Provide a link to the affected event from your Sentry account~

## Versions + Platform

- [x] SDK version - `@sentry/electron@v3.0.6`
- [x] Electron version - `electron@v14.2.7`
- [x] Platform - `macOS`

## Description

The following code throws a TypeError when called in renderer

```ts
import { init } from "@sentry/electron";

init({ dns: <valid dsn> });
```

The problem seems to be that `autoSessionTracking` is "on by default" but only works in `main` so it throws in `renderer.

**Comment by timfish** [comment]
When using the root export (ie. `import { init } from "@sentry/electron"`) the SDK determines which code to load at runtime but this sometime has incompatibilities with bundling.

Working around this is [covered in the docs](
https://docs.sentry.io/platforms/javascript/guides/electron/#bundler-configuration):

> ### Bundler Configuration
> The SDK attempts to ensure JavaScript bundlers pick up the correct entry point for the Electron process type, but this is not foolproof. If the incorrect entry point is selected, your bundler may throw an error or an error will be thrown at runtime. If this occurs you can use relative imports to ensure that the correct code is bundled for the process.
>
> In the main process:
>
>```ts
>import * as Sentry from "@sentry/electron/main";
> ```
> In the renderer processes:
> ```ts
> import * as Sentry from "@sentry/electron/renderer";
> ```

**Comment by Nokel81** [comment]
Yes I switched to that configuration, I just found it odd that `process?.type !== "browser"` doesn't work.

**Comment by timfish** [comment]
`process?.type !== "browser"` detection only works at runtime. 

If you're bundling your renderer code, the bundler doesn't know what `process.type` will be and so has to include both the main and renderer parts of the SDK and the SDK main code is not designed to be run in the browser.

**Comment by Nokel81** [comment]
No I know, but why is any code being run other than what `init` chooses to be run? It seems strange that the `main` part would be executed if `init` finds that it is in `renderer`.

**Comment by Nokel81** [comment]
Here is the stack trace:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'isReady')
    at Dev.js:125937
    at Module../node_modules/@sentry/electron/esm/main/electron-normalize.js (Dev.js:125944)
    at Module.options.factory (Dev.js:728368)
    at __webpack_require__ (Dev.js:727759)
    at fn (Dev.js:728039)
    at Module../node_modules/@sentry/electron/esm/main/context.js (Dev.js:125606)
    at Module.options.factory (Dev.js:728368)
    at __webpack_require__ (Dev.js:727759)
    at fn (Dev.js:728039)
    at Module../node_modules/@sentry/electron/esm/main/sdk.js (Dev.js:127753)
```

**Comment by Nokel81** [comment]
If using `await import(...)` works in my bundled code, shouldn't that also work in this library?

**Comment by timfish** [comment]
> Here is the stack trace

It look like the error is triggered by this code executing:
https://github.com/getsentry/sentry-electron/blob/75a3b3bbd0ce98f2506b74d3a2b3cf55b11ed2cf/src/main/electron-normalize.ts#L20-L28

I'm going to move this code:
```ts
import { ensureProcess } from '../common';
ensureProcess('main');
```
From `./src/main/index.ts` to `./src/main/sdk.ts`. This is where it should have been all along and would have at least given you an actionable error like this:
```
This code is intended to run in the Electron main process but is currently running in the renderer process.
This can occur if a bundler picks the wrong entry point.

You can work around this by using a relative import:
import * as Sentry from '@sentry/electron/renderer';
```

Do you not get any errors from webpack when bundling? What webpack `target` to you have on your config? I would expect webpack to complain about lots of things if you try and bundle this code for the browser.

> If using await import(...) works in my bundled code, shouldn't that also work in this library?

We would have to change every import in the main process code to be `async import()`. This would make the code much more complex and would affect performance simply to support loading the main process code into the renderer. And you really DON'T want the SDK main process code in your renderer bundle as it adds at least 200kB.

**Comment by timfish** [comment]
A fix for this has been released in 3.0.7
